doc/source/programs/gdal_raster_viewshed.rst

```diff
@@ -0,0 +1,169 @@
+   The magnitude of this effect varies with atmospheric conditions and depends on the wavelength.
+
+   Different applications for calculating visibility use different interchangeable notation to describe this phenomena:
+   Refraction Coefficient, Curvature Coefficient, and Sphere Diameter Factor.
+   gdal_viewshed uses the Curvature Coefficient notation.
+
+   .. math::
+
+     {CurvCoeff}=1-{RefractionCoeff}
+
+   Changes in air density curve the light downward causing an observer to see further and the earth to appear less curved,
+   as if the sphere (earth) diameter is larger then it actually is.
+   The ratio between that imaginary sphere diameter and the actual sphere diameter is given by the formula:
+
+   .. math::
+     {SphereDiameterFactor}=1/{CurvCoeff}=1/(1-{RefractionCoeff})
+
+   For visible light, the standard atmospheric refraction coefficient that is generally used is 1/7.
+   Thus the default value (since GDAL 3.4) for CurvCoeff that gdal_viewshed uses is 0.85714 (=~ 1-1/7)
+   for Earth CRS. Starting with GDAL 3.6, for non-Earth CRS (those whole semi-major axis differs
+   by more than 5% with the one of WGS 84), CurvCoeff default value is 1.0, to account for
+   the no refraction use case.
+
+   The height of the DEM is corrected according to the following formula:
+
+   .. math::
+
+      Height_{Corrected}=Height_{DEM}-{CurvCoeff}\frac{{TargetDistance}^2}{SphereDiameter}
+
+   Typical coefficient values are given in the table below (use Curvature Coeff value for the cc option)
+
+   ================  ==================  ===================  =====================
+   Use Case          Refraction Coeff    **Curvature Coeff**  Sphere Diameter Factor
+   No Refraction     0                   1                    1
+   Visible Light     1/7                 6/7 (=~0.85714)      7/6 (=~1.1666)
+   Radio Waves       0.25 ~ 0.325        0.75 ~ 0.675         1.33 ~ 1.48
+   Flat Earth        1                   0                    inf
+   ================  ==================  ===================  =====================
+
+.. option:: --invisible-value <value>
+
+   Pixel value to set for invisible areas. (Not supported in cumulative mode) Default: 0
+
+.. option:: --out-of-range-value <value>
+
+   Pixel value to set for the cells that fall outside of the range specified by
+   the observer location and the maximum distance. (Not supported in cumulative mode) Default: 0
+
+.. option:: --visible-value <value>
+
+   Pixel value to set for visible areas. (Not supported in cumulative mode) Default: 255
+
+.. option:: --mode normal|DEM|ground|cumulative
+
+   Sets what information the output contains.
+
+   - ``normal`` (the default) returns a raster of type Byte containing visible locations.
+
+   - ``DEM`` and ``ground`` return a raster of type Float64 containing the minimum target
+     height for target to be visible from the DEM surface or ground level respectively.
```

Comment:
What is the difference between DEM surface and ground level?

Reply:
I've added the following:
"""
  That is to say, if the minimum target height for the target to be visible at a
  point is ``h`` and the value of the input raster at that point is ``E``,
  for ``DEM``, ``E + h`` will be the output value.
  For ``ground``, ``h`` will be output value.
"""